### PR TITLE
Further optimize the performance

### PR DESCRIPTION
**Pull request by Wu-Chenyang**
Changes include:
- remove packages irrelevant to the domain (How about renaming this package to be something like `Roomba.jl`?)
- make the state type of the discrete roomba to be `Int`, such that it can be solved by `SARSOP.jl` (SARSOP involves an operation to check if two states are the same. However, the equivalence of `RoombaState` depends on the specific discretization.)
- rewrite the collision detection code to make it faster

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -2,45 +2,36 @@ name = "AA228FinalProject"
...
 POMCPOW = "4c53ee00-974c-466f-8fa5-8dd73959bbab"
 POMDPModelTools = "08074719-1b2a-587c-a292-00f91cc44415"
-POMDPPolicies = "182e52fb-cfd0-5e46-8c26-fd0667c990f4"
-POMDPSimulators = "e0d0a172-29c6-5d4e-96d0-f262df5d01fd"
-POMDPSolve = "a575d7f4-a5df-53f5-8ba4-9d691c1ba8ff"
 POMDPs = "a93abf59-7444-517b-a68a-c42f96afdd7d"
 Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
 ParticleFilters = "c8b314e2-9260-5cf8-ae76-3be7461ca6d0"
-QMDP = "3aa3ecc9-5a5d-57c8-8188-3e47bd8068d2"
+ProfileView = "c46f51b8-102a-5cf2-8d2c-8597cb0e0da7"
```
Comment (mykelk) [inline Project.toml]: Why is ProfileView added here?

**Review by mykelk — CHANGES_REQUESTED** [review]

**Comment by mykelk** [comment]
Thanks for the PR! I'm hoping @kunalmenda can do a quick review and then we can merge.

File: Project.toml (hunk 1 above)
Comment (Wu-Chenyang) [reply Project.toml]: It was added for debugging. I will remove it.

**Comment by zsunberg** [comment]
This is very cool, Thanks! I might be able to do a review tomorrow. I think a rename of the package would be in order. If you submit a PR to rename the module RoombaPOMDPs.jl, I will accept it and remove the package (I think Roomba might be too generic of a name - that sounds like it could be an interface for controlling a roomba).

**Comment by mykelk** [comment]
Yes, good idea to rename to RoombaPOMDPs.jl.

File: test/runtests.jl [hunk 2]
```diff
@@ -83,4 +82,31 @@ for (ext, mime) in ["html"=>MIME("text/html"), "svg"=>MIME("image/svg+xml"), "pn
         show(f, mime, v)
     end
     @test filesize(fname) > 0
+end
+
+m = RoombaPOMDP(sensor=sensor, mdp=RoombaMDP(config=config, aspace=vec([RoombaAct(v, om) for v in range(0, stop=2, length=2) for om in range(-2, stop=2, length=3)]), sspace=DiscreteRoombaStateSpace(41, 26, 20)));
+
+@testset "DiscreteRoomba Test" begin
```
Comment (zsunberg) [inline test/runtests.jl]: Would POMDPTesting.has_consistent_distributions work here?

**Review by zsunberg — APPROVED** [review]
I did not closely look at everything, but overall I think it looks good after you fix Mykel's suggestion.

File: test/runtests.jl (hunk 2 above)
Comment (Wu-Chenyang) [reply test/runtests.jl]: It does work.

**Comment by Wu-Chenyang** [comment]
I have made changes according to the suggestions mentioned. :)

**Review by mykelk — APPROVED** [review]

**Comment by mykelk** [comment]
Thanks! Would you mind addressing #11 ?

**Comment by Wu-Chenyang** [comment]
I will address it in the next several days.

**Comment by mykelk** [comment]
Thanks so much!

**Comment by zsunberg** [comment]
Thanks a bunch! This is great!

**Pull request merged**